src/catch2/internal/catch_compiler_detections.hpp

```diff
@@ -0,0 +1,26 @@
+
+//              Copyright Catch2 Authors
+// Distributed under the Boost Software License, Version 1.0.
+//   (See accompanying file LICENSE_1_0.txt or copy at
+//        https://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef CATCH_COMPILER_DETECTIONS_HPP_INCLUDED
+#define CATCH_COMPILER_DETECTIONS_HPP_INCLUDED
+
+#if defined(__GNUC__) && !defined(__clang__) && !defined(__ICC) && !defined(__CUDACC__) && !defined(__LCC__)
+#define CATCH_COMPILER_GCC
+#endif
+
+#if defined(__clang__)
```

Comment:
This definitely doesn't filter out all the different compilers that like to pretend they are Clang, e.g. IBM XL. See above for a good way to fix this.